pkg/pillar/nireconciler/linux_config.go

```diff
@@ -1259,13 +1279,16 @@ func (r *LinuxNIReconciler) getIntendedDnsmasqCfg(niID uuid.UUID) (items []dg.It
 	// DNS server configuration
 	var listenIP net.IP
 	if bridgeIP != nil {
-		listenIP = bridgeIP.IP
+		listenIP = bridgeIP
 	}
 	dnsCfg := generic.DNSServer{
 		ListenIP: listenIP,
 	}
 	for _, port := range ni.bridge.Ports {
 		for _, dnsSrv := range port.DNSServers {
+			if dnsSrv.To4() == nil && !ni.config.IsIPv6() {
```

Comment:
We're skipping rows which can't be converted to ipv4 and our configuration is not ipv6, why?

Reply:
Huh, To4() == nil is not obvious indication that IP is v6, thanks for explanation!